File: Practica_1/src/gestorAplicacion/Usuario.java
package gestorAplicacion;
/*Clase Usuario se usa para crear los clientes que realizen reservas en el hotel */

import java.util.ArrayList;

public class Usuario extends Persona {
    String tipo;
    ArrayList<Factura> listaFacturas;

    // METODOS

    /*
     * Metod sobreescrito de Persona que devuelve la impormacion mas importante el
     * usurio
     */
    @Override
    public String informacion() {
        return ("Nombre: " + this.getNombre() + "\n" +
                "Telefono: " + this.getTelefono() + "\n" +
                "Documento: " + this.getIdentificacion() + "\n");
    }

    // CONSTRUCTOR
    public Usuario(String nombre, int identificacion, int telefono, String tipo) {
        super(nombre, identificacion, telefono);
        this.tipo = tipo;
    }

    // GETTERS AND SETTERS
    public String getTipo() {
        return tipo;
    }

    public void setTipo(String tipo) {
        this.tipo = tipo;
    }

    public ArrayList<Factura> getListaFacturas() {
        return listaFacturas;
    }

    public void setListaFacturas(ArrayList<Factura> listaFacturas) {
        this.listaFacturas = listaFacturas;
    }

}

File: Python/gestorAplicacion/modelos/Usuario.py
from Persona import Persona

class Usuario(Persona):
    
    def __init__(self, nombre, identificacion, telefono, tipo, cuenta_bancaria):
        super().__init__(nombre, identificacion, telefono)
        self.tipo = tipo
        self.cuenta_bancaria = cuenta_bancaria
        self.lista_facturas = []

    def informacion(self):
        return f"Nombre: {self.nombre}\nTelefono: {self.telefono}\nDocumento: {self.identificacion}"

    def get_tipo(self):
        return self.tipo

    def set_tipo(self, tipo):
        self.tipo = tipo

    def getLista_facturas(self):
        return self.lista_facturas

    def setLista_facturas(self, lista_facturas):
        self.lista_facturas = lista_facturas

    def get_cuenta_bancaria(self):
        return self.cuenta_bancaria

    def set_cuenta_bancaria(self, cuenta_bancaria):
        self.cuenta_bancaria = cuenta_bancaria
